tests/components/gree/test_climate.py

```diff
@@ -407,6 +414,58 @@ async def test_send_target_temperature(
         state.attributes.get(ATTR_CURRENT_TEMPERATURE)
         == fake_device.current_temperature
     )
+    assert state.state == HVAC_MODES.get(fake_device.mode)
+
+    # Reset config temperature_unit back to CELSIUS, required for
+    # additional tests outside this component.
+    hass.config.units.temperature_unit = UnitOfTemperature.CELSIUS
+
+
+@pytest.mark.parametrize(
+    ("units", "temperature", "hvac_mode"),
+    [
+        (UnitOfTemperature.CELSIUS, 26, HVACMode.OFF),
+        (UnitOfTemperature.CELSIUS, 26, HVACMode.HEAT),
+        (UnitOfTemperature.CELSIUS, 26, HVACMode.COOL),
+        (UnitOfTemperature.CELSIUS, 26, HVACMode.AUTO),
+        (UnitOfTemperature.CELSIUS, 26, HVACMode.DRY),
+        (UnitOfTemperature.CELSIUS, 26, HVACMode.FAN_ONLY),
+    ],
+)
+async def test_send_target_temperature_with_hvac_mode(
+    hass: HomeAssistant, discovery, device, units, temperature, hvac_mode
+) -> None:
+    """Test for sending target temperature command to the device alongside hvac mode."""
+    hass.config.units.temperature_unit = units
+
+    fake_device = device()
+    if units == UnitOfTemperature.FAHRENHEIT:
+        fake_device.temperature_units = 1
+
+    await async_setup_gree(hass)
+
+    # Make sure we're trying to test something that isn't the default
+    assert fake_device.current_temperature != temperature
```

Comment:
```suggestion
```
Not needed as it will test it regardless if trying to set the same temperature as already is.